Jetchat/app/src/main/java/com/example/compose/jetchat/components/AnimatingFabContent.kt

```diff
@@ -43,17 +42,57 @@ fun AnimatingFabContent(
     extended: Boolean = true
 ) {
     val currentState = if (extended) ExpandableFabStates.Extended else ExpandableFabStates.Collapsed
-    val transitionDefinition = remember { fabTransitionDefinition() }
-    val transition = transition(
-        definition = transitionDefinition,
-        toState = currentState
-    )
+    val transition = updateTransition(currentState)
+
+    val textOpacity by transition.animateFloat(
+        transitionSpec = {
+            if (targetState == ExpandableFabStates.Collapsed) {
```

Comment:
Nit: move the if to the call site of `delayMillis`?

i.e. `delayMillis = when (targetState)  // ...`

Reply:
I think this is slightly more readable and copypasteable (for a sample)